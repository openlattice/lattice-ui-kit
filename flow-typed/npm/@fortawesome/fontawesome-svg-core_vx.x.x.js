// flow-typed signature: 39887e5e6c6c32861704232d9719cd14
// flow-typed version: <<STUB>>/@fortawesome/fontawesome-svg-core_v1.2.4/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/fontawesome-svg-core'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/fontawesome-svg-core' {
  declare module.exports: any;
}
