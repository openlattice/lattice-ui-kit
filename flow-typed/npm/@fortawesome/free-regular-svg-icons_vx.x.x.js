// flow-typed signature: e06b1e176fa03c6cedb9f03bb6cc7872
// flow-typed version: <<STUB>>/@fortawesome/free-regular-svg-icons_v5.13.0/flow_v0.118.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/free-regular-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/free-regular-svg-icons' {
  declare module.exports: any;
}
