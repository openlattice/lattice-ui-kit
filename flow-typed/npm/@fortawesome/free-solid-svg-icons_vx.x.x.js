// flow-typed signature: 64f61153e222e791ad6eca6c60c55baa
// flow-typed version: <<STUB>>/@fortawesome/free-solid-svg-icons_v5.13.0/flow_v0.118.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/free-solid-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/free-solid-svg-icons' {
  declare module.exports: any;
}
