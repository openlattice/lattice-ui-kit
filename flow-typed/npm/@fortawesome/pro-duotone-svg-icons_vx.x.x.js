// flow-typed signature: dc91210f6aed06e504e8c5d05af6775b
// flow-typed version: <<STUB>>/@fortawesome/pro-duotone-svg-icons_v5.10.2/flow_v0.106.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/pro-duotone-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/pro-duotone-svg-icons' {
  declare module.exports: any;
}
