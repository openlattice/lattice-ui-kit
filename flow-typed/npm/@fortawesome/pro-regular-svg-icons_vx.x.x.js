// flow-typed signature: c41877c2f635a0484c823342d488e66e
// flow-typed version: <<STUB>>/@fortawesome/pro-regular-svg-icons_v5.3.1/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/pro-regular-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/pro-regular-svg-icons' {
  declare module.exports: any;
}
