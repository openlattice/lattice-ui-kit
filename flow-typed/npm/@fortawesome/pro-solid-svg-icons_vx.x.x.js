// flow-typed signature: b2f94000db186d565d3eb121a020bf27
// flow-typed version: <<STUB>>/@fortawesome/pro-solid-svg-icons_v5.3.1/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/pro-solid-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/pro-solid-svg-icons' {
  declare module.exports: any;
}
