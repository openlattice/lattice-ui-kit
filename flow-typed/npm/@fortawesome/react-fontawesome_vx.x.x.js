// flow-typed signature: 561f463b72d39e8e0057aaad26de937b
// flow-typed version: <<STUB>>/@fortawesome/react-fontawesome_v0.1.2/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/react-fontawesome'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/react-fontawesome' {
  declare module.exports: any;
}
