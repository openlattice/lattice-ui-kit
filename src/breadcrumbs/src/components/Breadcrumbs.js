// @flow
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import type { Node } from 'react';

import { faChevronRight } from '@fortawesome/pro-regular-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Breadcrumbs as MUIBreadcrumbs } from '@material-ui/core';

type Props = {|
  separator ?:Node;
|};

const Breadcrumbs = (props :Props) => <MUIBreadcrumbs {...props} />;

Breadcrumbs.defaultProps = {
  separator: <FontAwesomeIcon icon={faChevronRight} fixedWidth />
};

export default Breadcrumbs;
