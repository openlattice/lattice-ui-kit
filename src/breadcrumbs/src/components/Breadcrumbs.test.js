import React from 'react';

import toJson from 'enzyme-to-json';
import { faChevronRight } from '@fortawesome/pro-regular-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { mount } from 'enzyme';

import Breadcrumbs from './Breadcrumbs';

describe('Breadcrumbs', () => {

  test('should match snapshot', () => {
    const wrapper = mount(
      <Breadcrumbs>
        <span>Foo</span>
        <span>Bar</span>
        <span>Baz</span>
      </Breadcrumbs>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should render FontAwesomeIcon with faChevronRight as default separator', () => {
    const wrapper = mount(
      <Breadcrumbs>
        <span>Foo</span>
        <span>Bar</span>
        <span>Baz</span>
      </Breadcrumbs>
    );
    const separators = wrapper.find(FontAwesomeIcon);
    expect(separators).toHaveLength(2);
    separators.forEach((separator) => {
      expect(separator.prop('icon')).toEqual(faChevronRight);
    });
  });
});
