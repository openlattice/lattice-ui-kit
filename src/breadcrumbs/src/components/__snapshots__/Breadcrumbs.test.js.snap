// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should match snapshot 1`] = `
<Breadcrumbs
  separator={
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={true}
      flip={null}
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [],
            "f054",
            "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
          ],
          "iconName": "chevron-right",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  }
>
  <WithStyles(ForwardRef(Breadcrumbs))
    separator={
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={true}
        flip={null}
        icon={
          Object {
            "icon": Array [
              320,
              512,
              Array [],
              "f054",
              "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
            ],
            "iconName": "chevron-right",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    }
  >
    <ForwardRef(Breadcrumbs)
      classes={
        Object {
          "li": "MuiBreadcrumbs-li",
          "ol": "MuiBreadcrumbs-ol",
          "root": "MuiBreadcrumbs-root",
          "separator": "MuiBreadcrumbs-separator",
        }
      }
      separator={
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={true}
          flip={null}
          icon={
            Object {
              "icon": Array [
                320,
                512,
                Array [],
                "f054",
                "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
              ],
              "iconName": "chevron-right",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      }
    >
      <WithStyles(ForwardRef(Typography))
        className="MuiBreadcrumbs-root"
        color="textSecondary"
        component="nav"
      >
        <ForwardRef(Typography)
          className="MuiBreadcrumbs-root"
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter",
              "alignJustify": "MuiTypography-alignJustify",
              "alignLeft": "MuiTypography-alignLeft",
              "alignRight": "MuiTypography-alignRight",
              "body1": "MuiTypography-body1",
              "body2": "MuiTypography-body2",
              "button": "MuiTypography-button",
              "caption": "MuiTypography-caption",
              "colorError": "MuiTypography-colorError",
              "colorInherit": "MuiTypography-colorInherit",
              "colorPrimary": "MuiTypography-colorPrimary",
              "colorSecondary": "MuiTypography-colorSecondary",
              "colorTextPrimary": "MuiTypography-colorTextPrimary",
              "colorTextSecondary": "MuiTypography-colorTextSecondary",
              "displayBlock": "MuiTypography-displayBlock",
              "displayInline": "MuiTypography-displayInline",
              "gutterBottom": "MuiTypography-gutterBottom",
              "h1": "MuiTypography-h1",
              "h2": "MuiTypography-h2",
              "h3": "MuiTypography-h3",
              "h4": "MuiTypography-h4",
              "h5": "MuiTypography-h5",
              "h6": "MuiTypography-h6",
              "noWrap": "MuiTypography-noWrap",
              "overline": "MuiTypography-overline",
              "paragraph": "MuiTypography-paragraph",
              "root": "MuiTypography-root",
              "srOnly": "MuiTypography-srOnly",
              "subtitle1": "MuiTypography-subtitle1",
              "subtitle2": "MuiTypography-subtitle2",
            }
          }
          color="textSecondary"
          component="nav"
        >
          <nav
            className="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            <ol
              className="MuiBreadcrumbs-ol"
            >
              <li
                className="MuiBreadcrumbs-li"
                key="child-0"
              >
                <span
                  key=".0"
                >
                  Foo
                </span>
              </li>
              <li
                aria-hidden={true}
                className="MuiBreadcrumbs-separator"
                key="separator-0"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={true}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        320,
                        512,
                        Array [],
                        "f054",
                        "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                      ],
                      "iconName": "chevron-right",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-chevron-right fa-w-10 fa-fw "
                    data-icon="chevron-right"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </li>
              <li
                className="MuiBreadcrumbs-li"
                key="child-1"
              >
                <span
                  key=".1"
                >
                  Bar
                </span>
              </li>
              <li
                aria-hidden={true}
                className="MuiBreadcrumbs-separator"
                key="separator-1"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={true}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        320,
                        512,
                        Array [],
                        "f054",
                        "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                      ],
                      "iconName": "chevron-right",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-chevron-right fa-w-10 fa-fw "
                    data-icon="chevron-right"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </li>
              <li
                className="MuiBreadcrumbs-li"
                key="child-2"
              >
                <span
                  key=".2"
                >
                  Baz
                </span>
              </li>
            </ol>
          </nav>
        </ForwardRef(Typography)>
      </WithStyles(ForwardRef(Typography))>
    </ForwardRef(Breadcrumbs)>
  </WithStyles(ForwardRef(Breadcrumbs))>
</Breadcrumbs>
`;
